### ISSUE TITLE
Add the ability to project DOM VR content into a element

### ISSUE THREAD

**Issue by caseyyee** [vvuk/gecko-dev#25]
Implement CSS property to specify that a element is to be interpreted as a stereo left/right pair.

This will be used in cases where we want to create stereo rendered output outside of using requestFullscreen.

**Comment by vvuk** [comment]
Can you explain what use case this would have?

**Comment by vvuk** [comment]
Based on slack discussion: this is basically allowing an iframe (e.g. the horzion browser frame) to act as a "portal" into another 3D world/view, with the right things happening in each eye.  We can look into this, but it will be far down the line.

**Comment by brianchirls** [comment]
Yes, this would definitely be useful for that. Could also be used for stereoscopic images and videos.

**Comment by brianchirls** [comment]
As long as CSS support is coming back around, I'm strongly in favor of this (though I acknowledge it must be extremely challenging to implement). It could be the best bet to implement seamless transitions between Web VR sites, without having to completely re-think and re-implement the browser UI for VR. It would allow the community to experiment with interaction models before having to commit to one approach built-in to the browser.

I think this feature needs to be paired with a way for a given iframe to determine it's position/orientation relative to the top-level camera, so it can render accordingly. For CSS elements inside the iframe, it might be handled by the browser, but if you have a full-window canvas inside that iframe, you need a way to get the correct perspective.

**Comment by caseyyee** [comment]
Update:
This looks like it _kind of_ works.    We get a stereo view, but the perspective is way off:

![image](https://cloud.githubusercontent.com/assets/1560884/7874146/8167fdea-0560-11e5-89d9-49d3b8a78470.png)

Viewable example here:
http://caseyyee.github.io/experiment-cssvr-iframe/iframe.html

**Comment by caseyyee** [comment]
Filed in bugzilla https://bugzilla.mozilla.org/show_bug.cgi?id=1171764
